cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -3146,6 +3161,64 @@ String getInternalServiceHostname(String serviceName)    {
             return withVoid(Future.succeededFuture());
         }
 
+        Future<ReconciliationState> getJmxTransDescription() {
+            int numOfBrokers = kafkaCluster.getReplicas();
+            this.jmxTrans = JmxTrans.fromCrd(kafkaAssembly, versions);
+            if (this.jmxTrans != null) {
+                this.jmxTransConfigMap = jmxTrans.generateJmxTransConfigMap(kafkaAssembly.getSpec().getJmxTransSpec(), numOfBrokers);
+                this.jmxTransDeployment = jmxTrans.generateDeployment(imagePullPolicy, imagePullSecrets);
+            }
+
+            return Future.succeededFuture(this);
+        }
+
+        Future<ReconciliationState> jmxTransConfigMap() {
+            return withVoid(configMapOperations.reconcile(namespace,
+                    JmxTrans.jmxTransConfigName(name),
+                    jmxTransConfigMap));
+        }
+
+
+        Future<ReconciliationState> jmxTransServiceAccount() {
+            return withVoid(serviceAccountOperations.reconcile(namespace,
+                    JmxTrans.containerServiceAccountName(name),
+                    jmxTrans != null ? jmxTrans.generateServiceAccount() : null));
+        }
+
+        Future<ReconciliationState> jmxTransDeployment() {
+            if (this.jmxTrans != null && this.jmxTransDeployment != null) {
+                Future<Deployment> deploymentFuture = deploymentOperations.getAsync(namespace, this.jmxTrans.getName());
+                return deploymentFuture.compose(dep -> {
+                    Future<ConfigMap> configMapFuture = configMapOperations.getAsync(namespace, jmxTransConfigMap.getMetadata().getName());
+                    return configMapFuture.compose(res -> {
+                        String resourceVersion = res.getMetadata().getResourceVersion();
+                        // getting the current cluster CA generation from the current deployment, if it exists
+                        int caCertGeneration = getCaCertGeneration(this.clusterCa);
+                        Annotations.annotations(jmxTransDeployment.getSpec().getTemplate()).put(
+                                Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, String.valueOf(caCertGeneration));
+                        Annotations.annotations(jmxTransDeployment.getSpec().getTemplate()).put(
+                                JmxTrans.CONFIG_MAP_ANNOTATION_KEY, resourceVersion);
+                        return withVoid(deploymentOperations.reconcile(namespace, JmxTrans.jmxTransName(name),
```

Comment:
I think the `withVoid` does the same as the `.map(i -> this)` does. I think you can leave out one or another.

Reply:
Done!